src/main/java/net/earthcomputer/clientcommands/command/PingCommand.java

```diff
@@ -1,41 +1,80 @@
 package net.earthcomputer.clientcommands.command;
 
+import com.mojang.authlib.GameProfile;
 import com.mojang.brigadier.Command;
 import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
 import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.network.ClientPlayNetworkHandler;
 import net.minecraft.client.network.PlayerListEntry;
 import net.minecraft.text.Text;
 
+import java.util.Collection;
+
+import static dev.xpple.clientarguments.arguments.CGameProfileArgumentType.*;
 import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.*;
 
 public class PingCommand {
+
+    private static final SimpleCommandExceptionType PLAYER_IN_SINGLEPLAYER_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("commands.cping.singleplayer"));
+    private static final SimpleCommandExceptionType PLAYER_NOT_FOUND_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("commands.cping.playerNotFound"));
+
     public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
         dispatcher.register(literal("cping")
-            .executes(ctx -> printPing(ctx.getSource())));
+            .executes(ctx -> printPing(ctx.getSource()))
+            .then(argument("player", gameProfile())
+                .executes(ctx -> printPing(ctx.getSource(), getCProfileArgument(ctx, "player"))))
+        );
     }
 
-    private static int printPing(FabricClientCommandSource source) {
+    private static int printPing(FabricClientCommandSource source) throws CommandSyntaxException {
+        if (source.getClient().isInSingleplayer()) {
+            throw PLAYER_IN_SINGLEPLAYER_EXCEPTION.create();
+        }
+
         int ping = getLocalPing();
+        if (ping == -1) {
+            throw PLAYER_IN_SINGLEPLAYER_EXCEPTION.create();
+        }
 
-        if (ping == -1 || source.getClient().isInSingleplayer()) {
-            source.sendFeedback(Text.translatable("commands.cping.local"));
-        } else {
-            source.sendFeedback(Text.translatable("commands.cping.multiplayer", ping));
+        source.sendFeedback(Text.translatable("commands.cping.success", ping));
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static int printPing(FabricClientCommandSource source, Collection<GameProfile> profiles) throws CommandSyntaxException {
+        if (source.getClient().isInSingleplayer()) {
+            throw PLAYER_IN_SINGLEPLAYER_EXCEPTION.create();
+        }
+
+        if (profiles.size() != 1) {
+            throw PLAYER_NOT_FOUND_EXCEPTION.create();
```

Comment:
Or too many players?

Reply:
I figured in that case _the_ player would still not be found 😄.

Reply:
There is precedent in vanilla + clientcommands commands to have a separate case for too many players